Libraries/LibWeb/CSS/SelectorEngine.cpp

```diff
@@ -190,148 +190,174 @@ static inline bool matches_indeterminate_pseudo_class(DOM::Element const& elemen
     return false;
 }
 
-static inline Web::DOM::Attr const* get_optionally_namespaced_attribute(CSS::Selector::SimpleSelector::Attribute const& attribute, GC::Ptr<CSS::CSSStyleSheet const> style_sheet_for_rule, DOM::Element const& element)
+static inline Vector<DOM::Attr const*> collect_matching_attributes(CSS::Selector::SimpleSelector::Attribute const& attribute_selector, GC::Ptr<CSS::CSSStyleSheet const> style_sheet_for_rule, DOM::Element const& element)
```

Comment:
Returning a Vector of GC allocated stuff like this is a bit dangerous, though it doesn't look to be unsafe in this particular case. Though as an idea after auditing that, maybe this could instead be something like `for_each_matching_attribute`, which would avoid the Vector stored GC objects and also avoid making a Vector altogether?

Reply:
Yeah I wasn't comfortable with it... I hadn't thought of doing a callback, but that does sound nicer, and wouldn't need to allocate a vector either. I'll give that a shot!